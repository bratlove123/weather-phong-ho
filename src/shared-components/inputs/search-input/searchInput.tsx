import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch, faCircleNotch } from "@fortawesome/free-solid-svg-icons";
import React, { useState } from "react";
import styles from "./searchInput.module.scss";
import PropTypes from 'prop-types';

export const SearchInput = ({
  placeholder,
  onSubmitSearch,
  saving,
}: {
  placeholder: string;
  onSubmitSearch: Function;
  saving: boolean;
}) => {
  const [value, setValue] = useState("");

  return (
    <div className={styles.form}>
      <input
        className={styles.textbox}
        placeholder={placeholder}
        value={value}
        onKeyDown={(e) => {
          if (e.key === "Enter" && value !== "") {
            onSubmitSearch(value);
            setValue("");
          }
        }}
        onChange={(e) => setValue(e.target.value)}
        type="text"
      />
      <button
        className={styles.button}
        onClick={() => {
          if (value !== "") {
            onSubmitSearch(value);
            setValue("");
          }
        }}
      >
        {saving ? (
          <FontAwesomeIcon icon={faCircleNotch} className="fa-spin" />
        ) : (
          <FontAwesomeIcon icon={faSearch} />
        )}
      </button>
    </div>
  );
};

SearchInput.propTypes = {
  placeholder: PropTypes.string,
  saving: PropTypes.bool,
  onSubmitSearch: PropTypes.func
};
