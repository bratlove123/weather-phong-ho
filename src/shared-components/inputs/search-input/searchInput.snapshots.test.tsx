import React from "react";
import { SearchInput } from "./searchInput";
import renderer from "react-test-renderer";

it("sets search button icon to spinner when saving is true", () => {
  const tree = renderer.create(
    <SearchInput saving onSubmitSearch={jest.fn()} placeholder="test" />
  );

  expect(tree).toMatchSnapshot();
});


it("sets search button icon to search when saving is false", () => {
  const tree = renderer.create(
    <SearchInput saving={false} onSubmitSearch={jest.fn()} placeholder="test" />
  );

  expect(tree).toMatchSnapshot();
});