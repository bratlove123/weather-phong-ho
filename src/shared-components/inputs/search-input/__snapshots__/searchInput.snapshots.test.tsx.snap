// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets search button icon to search when saving is false 1`] = `
<div
  className="form"
>
  <input
    className="textbox"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
  <button
    className="button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-search fa-w-16 "
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
</div>
`;

exports[`sets search button icon to spinner when saving is true 1`] = `
<div
  className="form"
>
  <input
    className="textbox"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="test"
    type="text"
    value=""
  />
  <button
    className="button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-circle-notch fa-w-16 fa-spin"
      data-icon="circle-notch"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </button>
</div>
`;
